rust/aptos-indexer-transaction-stream/src/transaction_stream.rs

```diff
@@ -313,93 +309,150 @@ pub async fn create_fetcher_loop(
     request_ending_version: Option<u64>,
     auth_token: String,
     processor_name: String,
-    transaction_filter: crate::transaction_filter::TransactionFilter,
-    // The number of transactions per protobuf batch
+    transaction_filter: TransactionFilter,
     pb_channel_txn_chunk_size: usize,
-) {
-    info!(
-        processor_name = processor_name,
-        service_type = crate::worker::PROCESSOR_SERVICE_TYPE,
-        stream_address = indexer_grpc_data_service_address.to_string(),
-        start_version = starting_version,
-        end_version = request_ending_version,
-        "[Parser] Connecting to GRPC stream",
-    );
-    let mut response = get_stream(
-        indexer_grpc_data_service_address.clone(),
-        indexer_grpc_http2_ping_interval,
-        indexer_grpc_http2_ping_timeout,
-        indexer_grpc_reconnection_timeout_secs,
-        starting_version,
-        request_ending_version,
-        auth_token.clone(),
-        processor_name.to_string(),
-    )
-    .await;
-    let mut connection_id = match response.metadata().get(GRPC_CONNECTION_ID) {
-        Some(connection_id) => connection_id.to_str().unwrap().to_string(),
-        None => "".to_string(),
-    };
-    let mut resp_stream = response.into_inner();
-    info!(
-        processor_name = processor_name,
-        service_type = crate::worker::PROCESSOR_SERVICE_TYPE,
-        stream_address = indexer_grpc_data_service_address.to_string(),
-        connection_id,
-        start_version = starting_version,
-        end_version = request_ending_version,
-        "[Parser] Successfully connected to GRPC stream",
-    );
+    resp_stream: Option<Streaming<TransactionsResponse>>,
+    connection_id: Option<String>,
+    next_version_to_fetch: u64,
+    reconnection_retries: u64,
+    last_fetched_version: i64,
+    fetch_ma: MovingAverage,
+}
```

Comment:
Maybe this change doesn't belong in this PR but I'd like us to think about how we would actually configure this from a config file. Half the fields here are just configuration, perhaps they could be moved into a `TransactionStreamConfig` struct (which would configure serde derives, have default values, etc) and then this struct has a `config: TransactionStreamConfig` field. We could then use this at a higher level rather than repeating the same fields all over again in the worker config or whatever.

This makes it much easier to using in an SDK setting, whereas today all the configuration stuff like defaults are detached from the actual place they're used.

I sort of attempted this in my SDK crate ages ago (I think I sent you the branch), you could use that for inspo.